### PR TITLE
MBS-9909: Change reports to use React (2 - editors)

### PR DESCRIPTION
**Pull request by reosarevok**

### CODE REVIEW

File: root/report/LimitedEditors.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}: ReportDataT<ReportEditorT>) => (
+  <Layout fullWidth title={l('Beginner/limited editors')}>
+    <h1>{l('Beginner/limited editors')}</h1>
+
+    <ul>
+      <li>
+        {l('This report lists {url|beginner/limited editors}.',
+          {url: '/doc/How_to_Create_an_Account'})}
+      </li>
+      <li>{l('Total editors found: {count}', {__react: true, count: pager.total_entries})}</li>
```
Comment (mwiencek) [inline root/report/LimitedEditors.js]: rm -rf `__react: true`

File: root/report/components/EditorList.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          <th>{l('Website')}</th>
+          <th>{l('Email')}</th>
+          <th>{l('Bio')}</th>
+        </tr>
+      </thead>
+      <tbody>
+        {items.map((item, index) => {
+          { /* Hack to make this linkable with EditorLink */ }
+          { /* $FlowFixMe */ }
+          item.entityType = 'editor';
```
Comment (mwiencek) [inline root/report/components/EditorList.js]: There should be a simple way we can clean this all up: in lib/MusicBrainz/Server/Report/LimitedEditors.pm, just add an `inflate_rows` method that creates an `Entity::Editor` from each row. Right now it's unfortunate how this `ReportEditorT` is leaking to other parts of the codebase (EntityLink, entityHref) despite only being invented for one file. :/

Comment (reosarevok) [reply root/report/components/EditorList.js]: Changed this to make it less hacky, and to load Editor hopefully it looks good now!

File: root/report/components/EditorList.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          const editor = item.editor;
+          return (
+            <tr className={loopParity(index)} key={editor.name}>
+              <td>
+                <EditorLink editor={editor} />
+                {' ('}
+                <a href={'/admin/user/delete/' + editor.name}>{l('delete')}</a>
+                {')'}
+              </td>
+              <td>{editor.registration_date}</td>
```
[the remark this answers is not in the document]
Comment (reosarevok) [reply root/report/components/EditorList.js]: This is intended for admins, so I'm not sure it matters, but we could.

[the remark this answers is not in the document]
Comment (reosarevok) [reply root/report/components/EditorList.js]: Only issue is we need $c then - does that make things slower or anything? If not, we might as well, but if it does, then I'd skip it here :)

[the remark this answers is not in the document]
Comment (mwiencek) [reply root/report/components/EditorList.js]: Okay, never mind then. :)

File: lib/MusicBrainz/Server/Report/EditorReport.pm [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    my $editors = $self->c->model('Editor')->get_by_ids(
+        map { $_->{id} } @$items
+    );
+
+    return [
+        map +{
+            %$_,
+            editor => $editors->{ $_->{id} }
+        },
+            @$items
```
[the remark this answers is not in the document]
Comment (reosarevok) [reply lib/MusicBrainz/Server/Report/EditorReport.pm]: It's the same on every WhateverReport package, it seems - but I can. Do I do it in the others as well with a second commit?

[the remark this answers is not in the document]
Comment (mwiencek) [reply lib/MusicBrainz/Server/Report/EditorReport.pm]: Ah, weird. That worksforme if you feel like updating them all.

**Pull request merged**